### ISSUE TITLE
Give an Option to Stop Reviewing and Train again

### ISSUE THREAD

**Issue by Vaibhav-DTx** [asreview/asreview#516]
## Feature Request

**Is your feature request related to a problem? Please describe.**
I started reviewing, and after 5 irrelevant items, I am given a choice, whether to stop or continue. I select to continue, and select two more irrelevant items. At this time, I wish to stop reviewing further records, and start the training/learning process. However, there is no such option at this phase. I have to keep on labelling the items, until I get the choice again. Sometimes this choice appears after I have marked 10 irrelevant items, and sometimes only after 15 irrelevant items.

**Describe the solution you'd like**
It would be useful if there is an option to stop reviewing and execute the training, at any point after the first 5 irrelevant items (which is the 'base' requirement for a good training).

**Describe alternatives you've considered**
Another option could be to have an option or a parameter at the dashboard, where the user specifies the number of irrelevant items required. Once this number is achieved, the training may start (or the option to start training is enabled). 

**Teachability, Documentation, Adoption, Migration Strategy**

**Comment by MartijnUX** [comment]
Yes! @Vaibhav-DTx You are right. As an user you should be able to stop anytime after selecting enough items. For this version of ASReview we thought that the "x" would be clear enough for the user to stop. When the user stops, you can see the number of articles changed in the overview panel.

![image](https://user-images.githubusercontent.com/45943709/111476081-12e25400-872e-11eb-8dd3-b0627ec9e8d4.png)

Would it be helpful for you when we would add an "Stop" in the left bottom of this panel? Short term solution (easiest solution I think @terrymyc ?). The long term solution is would be to create an progressbar/complete component within this panel.

<img width="459" alt="image" src="https://user-images.githubusercontent.com/45943709/111476273-4c1ac400-872e-11eb-9a36-1690e7b378a1.png">

**Comment by Vaibhav-DTx** [comment]
Hi @MartijnUX! Thank you very much for replying with detailed answer and possibilities. 
Yes, you're right, that "X" is good, but it does not clearly mean "stop labelling and move to the next phase". It may mean like "cancel". I was ignoring that for this reason. 
So I would definitely upvote for the solution that you have suggested. If I may suggest one step further, and instead of "STOP", may be use "FINISH" or "FINISHED", or perhaps something more descriptive like "Exit Oracle Mode".

**Comment by Rensvandeschoot** [comment]
> Furthermore, I thought, once I label a few more samples, I would be allowed to retrain the model.
The model is re-trained after each labelling decision (unless you click faster than the training takes, then you will see the second most relevant record to label) and the new most relevant record is displayed.

**Comment by Vaibhav-DTx** [comment]
Ok, that sounds great! 
Initially I thought it might be doing so, but it was so fast, I didn't realize it was training after each labelling decision. If I understand correctly, each new record displayed for labelling is the one that the system things is the next most relevant record, right? 
This means, when I download results, I am already downloading the newly trained (updated) results dataset. 
If so, this answers my question. I think this should be highlighted properly in the interface/messages. 
Thank you!

**Comment by Rensvandeschoot** [comment]
> it was so fast, I didn't realize it was training after each labelling decision

Yes indeed, and all credits to our software engineers for this!!!! 

> each new record displayed for labelling is the one that the system things is the next most relevant record,

Just note that only when you select `certainty based` sampling you will be shown the most relevant record from the pool. see our [documentation](https://asreview.readthedocs.io/en/latest/features/pre_screening.html#select-model) 

> This means, when I download results, I am already downloading the newly trained (updated) results dataset.

If you download the results, the information in the `csv` or `excel` file are from the most recent iteration of the model. Information from other iterations is stored in the project files `.asreview` which you also can [export](https://asreview.readthedocs.io/en/latest/features/post_screening.html#export-project) and ideally publish together with your paper for complete transparency. 


> I think this should be highlighted properly in the interface/messages.

Yes, this might be helpful indeed! Do you have a suggestion for the documentation how to make it clearer? Maybe [here](https://asreview.readthedocs.io/en/latest/features/screening.html)? If you have a good idea, please add text in the raw file on Github and start a Pull Request: https://github.com/asreview/asreview/blob/master/docs/source/features/screening.rst . Would be highly appreciated!!

**Comment by Rensvandeschoot** [comment]
@MartijnUX Maybe it is a good idea to start a new issue how we can visualize in the front-end the model is training while you are reading the abstract and whenever a new model is ready the ranking of abstracts is updated. I know you have already been thinking about it. So, let's put it in the open and ask our community for some help!

**Comment by Rensvandeschoot** [comment]
@terrymyc Do you think it is possible to implement such a 'stop' button in the prior selection panel?

Implementing a 'stop' button in the labelling panel requires more discussion and good thinking... we already have art-work for it: https://github.com/asreview/asreview-artwork/tree/master/ElasAdventures/5.HoldingSIGNS/PNG/200px

**Comment by MartijnUX** [comment]
<img width="178" alt="image" src="https://user-images.githubusercontent.com/45943709/111603414-9eaebb80-87d4-11eb-9c0e-dd7ff499710f.png">
If we look at the Download functionality, they are situated in multiple places within ASReview. During the multiple Usability Tests we didn't see any users having problems with that. Looking at it again, I think it is a good idea to give the user the possibility, while screening, to have the "finish/stop screening" option. That doesn't mean the user is going to download results, but just finishes screening (for this project) and goes back to the project dashboard. 
Thank you @Vaibhav-DTx for your input! We will improve with our users.

**Comment by terrymyc** [comment]
I think we lack an intuitive instruction on how to use ASReview. Although we have a rich documentation about every single element, no one really wants to read a detailed user manual, let alone a little tip buried somewhere. In this case, we gave the user (also ourselves) a feeling that the model was training during the prior knowledge labelling. However, in the setup stage, labelling prior knowledge won't train the model, meaning that "stop/resume" options can't be available. The model starts training only after clicking "Finish" in the last step.

Therefore, you don't actually start reviewing in the labelling prior knowledge step. In this step, the documents presented to you are randomly selected ("Random" button means that). You just need to provide at least 1 relevant and 1 irrelevant records to give the model sufficient knowledge to start. Once you click "Finish", you will be able to see "Start reviewing". Then the model is ready to help you!

The choice of stopping or showing more in the labelling prior knowledge step will pop up after labelling every 5 random irrelevant records. It's just a reminder of sufficient irrelevant records for model training. However, this causes the confusion.

I would suggest we may need a real "quick tour" to guide users through the process of creating, setting up, starting reviewing, finishing a project. This could be realised in two ways:
- A guided tour of the app (eg. https://shepherdjs.dev). Technically it may require a lot work.
- A video demonstration with text explanation for every step (maybe also euphonious bgm). Considering the fast iteration of ASReview, this may only be done for major releases.

For now, the easiest way to avoid confusion is to simply improve the texts in the prior knowledge labelling stage (eg. discard using the word "stop").

@Rensvandeschoot I'd say a "stop" button after starting reviewing is more or less a user experience issue. Technically the model resumes/stops training after every labelling decision. When "stop" means finishing a project, we have an option in the project dashboard to mark a project as finished (also the option to undo). The important thing would be how we can convey this message to users.

@Vaibhav-DTx Thanks for reporting this issue! We will improve the user experience.
